src/pyodide.js

```diff
@@ -331,13 +331,20 @@ var languagePluginLoader = new Promise((resolve, reject) => {
   Module.preloadedWasm = {};
   let isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
 
+  let wasm_fetch = fetch(wasmURL);
+  const compileBuffer = () =>
+      wasm_fetch.then(response => response.arrayBuffer())
+          .then(bytes => WebAssembly.compile(bytes));
   let wasm_promise;
   if (WebAssembly.compileStreaming === undefined) {
-    wasm_promise = fetch(wasmURL)
-                       .then(response => response.arrayBuffer())
-                       .then(bytes => WebAssembly.compile(bytes));
+    wasm_promise = compileBuffer();
   } else {
-    wasm_promise = WebAssembly.compileStreaming(fetch(wasmURL));
+    wasm_promise = WebAssembly.compileStreaming(wasm_fetch);
+    wasm_promise = wasm_promise.catch(e => {
+      if (e instanceof TypeError)
```

Comment:
I guess the wrong MIME type is not the only reason for a type error during streaming compilation? Maybe we should at least write a warning here to console, so that this doesn't happen silently?

Reply:
I'm not sure I follow that reasoning. If it's not due to the MIME type, I'd expect the same error to be output from the fallback path. I added a warning anyway; I'd like to know about it if my server is misconfigured.

Reply:
Oops, wait a second. `console.warn` is non-standard. I changed it to `console.error`. (invoking `(console.warn || console.error)` for this seemed out of place.)